### PR TITLE
Example MVP PWA

### PR DESCRIPTION
**Pull request by sjmiles**
Quick PWA template for projects that don't need full PWA Toolkit infrastructure or prefer a smaller footprint.

Depends on Xen library only (can remove this dependency too, if it ever matters).

An automated build would need to update the cache version number [here](https://github.com/PolymerLabs/arcs/pull/2348/commits/3c06812ef40501e274b18121f317bc49177ff6c8#diff-22ea706e2a7241871f6d406eef73629bR3) to allow clients to cache newer files.

@cromwellian I'm not trying to push this on you at all, continue to use your preferred tool. Rather, it was bothering me that I bloviated about it being _easy_ to roll your own PWA, when that's only true when you have already spent a lot of time on it like I have. I wanted to get this down to put my money where my mouth is.

### CODE REVIEW

**Comment by sjmiles** [comment]
Added more reviewers as nobody really has time to study this code. I believe it's only additive (capturing some tech) and won't hurt anything. I don't want to let it age too much.

File: modalities/dom/xen-pwa/pwa/service-worker.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      this.register(href, scope || '.');
+    }
+  }
+  register(href, scope) {
+    if ('serviceWorker' in navigator) {
+      navigator.serviceWorker.register(href, {scope: scope});
+    }
+  }
+}
+customElements.define('service-worker', ServiceWorker);
```
Comment (shaper) [inline modalities/dom/xen-pwa/pwa/service-worker.js]: This looks like a copy of the same ServiceWorker class in `modalities` dir above, is it intentional?

File: shells/single-shell/index.html [hunk 2]
```diff
@@ -32,6 +35,7 @@
 
   const env = new Env('../../');
   Object.assign(env.pathMap, {
+    'https://$shell/': `../../shells/`,
```
Comment (shaper) [inline shells/single-shell/index.html]: $shell is singular but the mapping is plural, and this differs from both existing cases below. Just making sure this is intentional.

Comment (sjmiles) [reply shells/single-shell/index.html]: Yes, this is a backward compatibility thing. Ty.

File: modalities/dom/xen-pwa/pwa/service-worker.js (hunk 1 above)
Comment (sjmiles) [reply modalities/dom/xen-pwa/pwa/service-worker.js]: Good catch. I wasn't sure how to handle it. 

I want pwa to be self-contained but I also want to share service-worker.js.

**Pull request merged**